backend/src/services/exercice/data.js

```diff
@@ -3,12 +3,38 @@ const Exercice = require('./model');
 module.exports = {
   create: async (request, exam) => {
     const exerciceToSave = new Exercice(request);
+    const maxOrder = Math.max(...exam.exercices.map(ex => ex.order));
+    exerciceToSave.order = exerciceToSave.order || maxOrder + 1;
     const exercice = await exerciceToSave.save();
     exam.exercices.push(exercice._id);
     await exam.save();
     return exercice;
   },
-  getById: async id => Exercice.findById(id),
+  getById: async id => Exercice.findById(id).populate('questions'),
   update: async (_id, exercice) => Exercice.findOneAndUpdate({ _id }, exercice),
   delete: async _id => Exercice.findByIdAndDelete(_id),
+
+
+  /**
+   * Updates the order of the other exercices that are contained
+   * in the same exam if it is needed
+   * @param {Exam} exam the parent exam that contains the exercices
+   * @param {Exercice} exercice the exercice that's being dragged and dropped
+   * @param {number} newExerciceOrder the new order of the exercice that's being dragged and dropped
+   */
+  async updateOrderOfOtherExercicesIfNeeded(exam, exercice, newExerciceOrder) {
```

Comment:
Why not just call it updateOrder or something like that ? We can comment or put somewhere that it's updating only if needed

Reply:
I wanted the name of the function to be very explicit, but yeah I can change if you prefer
@FelixDelcourt @bdebever @ArthurWirbel what do you prefer ?

Reply:
A bit "overkill" as startupers would say :rocket:

Reply:
I changed the name 👍